mesh/occupations/views.py

```diff
@@ -1,3 +1,166 @@
-from django.shortcuts import render
+# Django Imports
+from django.http import JsonResponse, HttpResponse
+from django.views import View
+from django.core import serializers
 
-# Create your views here.
+# Model Imports
+from .models import Occupation, OccupationBridge
+from mesh.accounts.models import Account
+from mesh.profiles.models import Profile
+
+# Utilities Imports
+from mesh.utils.validate_data import validate_json_and_required_fields
+from mesh.exceptions.MissingRequiredFields import MissingRequiredFields
+from mesh.exceptions.InvalidJsonFormat import InvalidJsonFormat
+
+# Library Imports
+import json
+
+
+class OccupationsView(View):
+    """
+        Handles HTTP requests related to the collection of Occupation,
+        supporting GET to retrieve the list of all Occupations and
+        POST to create a new Occupation.
+    """
+    
+    def get(self, request, *args, **kwargs):
+        """
+            Handle GET requests.
+
+            Returns a JSON response containing all Occupations.
+        """
+        occupations = Occupation.objects.all()
+        
+        # Create a list of dictionaries containing occupation details
+        occupations_list = serializers.serialize("json", occupations)
+        
+        return JsonResponse(occupations_list, status = 200, safe = False)
+    
+    def post(self, request, *args, **kwargs):
```

Comment:
please correct me if I'm wrong but the rest standard enforces a strict endpoint policy. So post should not be in charge of both creation and update, but rather the put's job.

Reply:
No you're absolutely right, I meant to remove it when I created the patch endpoint but it must've slipped my mind. Thank you for catching that!